voice/src/main/kotlin/gateway/Ticker.kt

```diff
@@ -0,0 +1,49 @@
+package dev.kord.voice.gateway
+
+import io.ktor.util.*
+import kotlinx.coroutines.*
+import kotlinx.coroutines.channels.ReceiveChannel
+import kotlinx.coroutines.channels.consumeEach
+import kotlinx.coroutines.channels.ticker
+import kotlinx.coroutines.sync.Mutex
+import kotlinx.coroutines.sync.withLock
+import mu.KotlinLogging
+import kotlin.coroutines.CoroutineContext
+
+private val logger = KotlinLogging.logger { }
+
+/**
+ * A reusable fixed rate ticker.
+ *
+ * @param dispatcher The dispatchers the events will be fired on.
+ */
+@ObsoleteCoroutinesApi
+class Ticker(private val dispatcher: CoroutineDispatcher = Dispatchers.Default) : CoroutineScope {
+    override val coroutineContext: CoroutineContext = SupervisorJob() + dispatcher + CoroutineName("Ticker")
+
+    private val mutex = Mutex()
+
+    private var ticker: ReceiveChannel<Unit>? = null
+
+    suspend fun tickAt(intervalMillis: Long, block: suspend () -> Unit) {
+        stop()
+        mutex.withLock {
```

Comment:
why are we using a mutex here?

Reply:
I copied this from the gateway Ticker and didn't think changing one but not the other was the right move.